server/gameObjects/Player.mjs

```diff
@@ -402,8 +402,13 @@ export default class Player {
   }
 
   canGetResource(energyCost) {
-    // @TODO: send message that action can't be performed #164
-    return this.isInRange(1) && this.energy >= energyCost;
+    if (!this.isInRange(1)) {
+      return "not-in-range";
```

Comment:
Don't use strings like that - those are keys of Messages object. 
Put the Messages in `./shared` and use on the client and on the server.